include/hybridmap/HybridNeutralAtomMapper.hpp

```diff
@@ -0,0 +1,534 @@
+   */
+  WeightedSwaps getExactSwapsToPosition(const Operation* op, HwQubits position);
+
+  // Cost function calculation
+  /**
+   * @brief Calculates the distance reduction for a swap gate given the
+   * necessary close by swaps and exact moves.
+   * @details Close by swaps are from two qubit gates, which only require to
+   * swap close by. The exact moves are from multi-qubit gates, that require
+   * swapping exactly to the multi-qubit gate position.
+   * @param swap The swap gate to compute the distance reduction for
+   * @param swapCloseBy The close by swaps from two-qubit gates
+   * @param moveExact The exact moves from multi-qubit gates
+   * @return The distance reduction cost
+   */
+  fp swapCostPerLayer(const Swap& swap, const Swaps& swapCloseBy,
+                      const WeightedSwaps& swapExact);
+  /**
+   * @brief Calculates the cost of a swap gate.
+   * @details The cost of a swap gate is computed with the following terms:
+   * - distance reduction for front + lookahead layers using swapCostPerLayer
+   * - decay term for blocked qubit from last swaps
+   * The cost is negative.
+   * @param swap The swap gate to compute the cost for
+   * @return The cost of the swap gate
+   */
+  fp swapCost(const Swap&                            swap,
+              const std::pair<Swaps, WeightedSwaps>& swapsFront,
+              const std::pair<Swaps, WeightedSwaps>& swapsLookahead);
+  /**
+   * @brief Calculates the cost of a move operation.
+   * @details Assumes the move is executed and computes the distance reduction
+   * for the layer.
+   * @param move The move operation to compute the cost for
+   * @param layer The layer to compute the distance reduction for
+   * @return The distance reduction cost
+   */
+  fp moveCostPerLayer(const AtomMove& move, GateList& layer);
+
+  /**
+   * @brief Calculates a parallelization cost if the move operation can be
+   * parallelized with the last moves.
+   * @param move The move operation to compute the cost for
+   * @return The parallelization cost
+   */
+  fp parallelMoveCost(const AtomMove& move);
+  /**
+   * @brief Calculates the cost of a move operation.
+   * @details The cost of a move operation is computed with the following terms:
+   * - distance reduction for front + lookahead layers using moveCostPerLayer
+   * - parallelization term based on last moves using parallelMoveCost
+   * The three contributions are weighted with the runtime parameters.
+   * @param move The move operation to compute the cost for
+   * @return The cost of the move operation
+   */
+  fp moveCost(const AtomMove& move);
+  /**
+   * @brief Calculates the cost of a series of move operations by summing up the
+   * cost of each move.
+   * @param moveComb The series of move operations to compute the cost for
+   * @return The total cost of the series of move operations
+   */
+  fp moveCostComb(const MoveComb& moveComb);
+
+  // temp
+  void printLayers();
```

Comment:
temp? worth fixing?

Reply:
In the beginning I thought it might only be temporary, therefore the comment. I now added a docstring and included it as a regular function, as it is helpful for debugging.